src/server/page_server.cpp

```diff
@@ -1021,9 +1025,10 @@ page_server::execute_catchup (cubthread::entry &entry, const LOG_LSA catchup_lsa
       assert (remaining_page_cnt == 0);
       _er_log_debug (ARG_FILE_LINE, "[CATCH_UP] The catch-up is completed, ranging from %lld to %lld, count = %lld.\n",
 		     start_pageid, end_pageid, total_page_count);
-      // TODO: send CATCHUP_DONE_MSG to the ATS
       // TODO: start appneding log prior nodes from the ATS.
 
+      push_request_to_active_tran_server (page_to_tran_request::SEND_CATCHUP_COMPLETE, std::string());
```

Comment:
Even in a situation where an error occurs during the catch-up process and it seems like the catch-up will stop,
wouldn't it be better to inform ATS of both the completion of the catch-up and the situation where it stops ?

Reply:
Do you mean that where the catch-up fails for some reason? It will be addressed in a TODO issue when the way to re-request comes in. And about were to start the catch-up again, it doesn't need to be stored or notified to the ATS since the ATS just sends the latest catchup_lsa and the PS starts the catch_up from the `append_lsa`, which is the point the previous catch-up stops.

Reply:
OK, I got it